waku-bindings/src/node/store.rs

```diff
@@ -0,0 +1,145 @@
+//! Waku store protocol related methods
+
+// std
+use std::ffi::CString;
+// crates
+use libc::*;
+use std::sync::Arc;
+use tokio::sync::Notify;
+// internal
+use crate::general::{
+    contenttopic::WakuContentTopic, messagehash::MessageHash, pubsubtopic::PubsubTopic, Result,
+    WakuStoreRespMessage,
+};
+use crate::node::context::WakuNodeContext;
+use crate::utils::{get_trampoline, handle_response, LibwakuResponse, WakuDecode};
+use multiaddr::Multiaddr;
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct PagingOptions {
+    pub page_size: usize,
+    pub cursor: Option<MessageHash>,
+    pub forward: bool,
+}
+
+/// Criteria used to retrieve historical messages
+#[derive(Clone, Serialize, Debug)]
+struct StoreQueryRequest {
+    /// if true, the store-response will include the full message content. If false,
+    /// the store-response will only include a list of message hashes.
+    request_id: String,
+    include_data: bool,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pubsub_topic: Option<PubsubTopic>,
+    content_topics: Vec<WakuContentTopic>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    time_start: Option<usize>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    time_end: Option<usize>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    message_hashes: Option<Vec<MessageHash>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pagination_cursor: Option<MessageHash>, // Message hash (key) from where to start query (exclusive)
+    pagination_forward: bool,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pagination_limit: Option<u64>,
+}
+
+#[derive(Clone, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct StoreWakuMessageResponse {
+    pub message_hash: MessageHash,
+    pub message: WakuStoreRespMessage,
```

Comment:
IIRC, `include_data` will determine whether a message will be returned or not, so probably makes sense for `message` to be an option

Reply:
Yes correct :) We are using the `include_data` concept in the requests. I think is fine unless I'm missing something

Reply:
But wouldn't that mean that if you send `include_data`: false,  your message will contain a (invalid) struct with default values? having it as an option perhaps would make it clearer for the dev that there's no message available